File: client/android/app/src/main/java/app/mobiling/client/notification/NotificationFeatureBridge.kt
package app.mobiling.client.notification

import app.mobiling.client.contract.notification.NotificationInboxPayload
import app.mobiling.client.contract.notification.NotificationSubscriptionRequest
import app.mobiling.client.data.notification.NotificationGateway

class NotificationFeatureBridge(
    private val gateway: NotificationGateway,
) {
    suspend fun inbox(): NotificationInboxPayload = gateway.inbox()
    suspend fun unreadCount(): Int = gateway.unreadCount()
    suspend fun markRead(ids: List<String>): Int = gateway.markRead(ids)
    suspend fun subscription(request: NotificationSubscriptionRequest): Boolean = gateway.subscription(request)
}

File: client/ios/App/Notification/NotificationFeatureBridge.swift
import Foundation

public struct NotificationFeatureBridge: Sendable {
    private let gateway: NotificationGateway

    public init(gateway: NotificationGateway) { self.gateway = gateway }
    public func inbox() async throws -> NotificationInboxPayload { try await gateway.inbox() }
    public func unreadCount() async throws -> Int { try await gateway.unreadCount() }
    public func markRead(ids: [String]) async throws -> Int { try await gateway.markRead(ids: ids) }
    public func subscription(request: NotificationSubscriptionRequest) async throws -> Bool { try await gateway.subscription(request: request) }
}
